src/store/index.ts

```diff
@@ -81,27 +81,33 @@ export const useStore = create<State>()((set, get) => ({
     );
     try {
       const api: WalletApi = await (window as any).cardano[walletName].enable();
-      const [rawAddress] = await api.getUnusedAddresses();
-      const address = fromHex(rawAddress);
-      const balance = await api.getBalance();
 
-      const decodedBalance = cbor.decode(fromHex(balance));
+      let addresses = await api.getUnusedAddresses();
+      if(addresses.length == 0)
+        addresses = await api.getUsedAddresses();
+      
+      const address = fromHex(addresses[0]);
       const words = bech32.toWords(address);
 
       const bechAddr = bech32.encode(
         address[0] == NetworkId.MAINNET ? 'addr' : 'addr_test',
         words,
         130
       );
-
+      let decodedBalance = 0;
+      try{
```

Comment:
If `getBalance` doesn't work on a wallet, then it's the wallet's problem and should not be accounted for here

Reply:
If we do not handle this here, the code becomes breaking for that particular wallet. I think we should remove it from the supported wallet.

Reply:
@muhammadumer-sl2 

What was the problem you experienced here with nami? `getBalance` seem to work for me (at least in the sense that it doesn't throw any errors).

This:

```ts
const [lovelacebalance] = cbor.decode(fromHex(balance))
```

works, but I'm not sure if it's logically correct. Seems weird that this would be the solution, because it seems to be non backward compatible.